src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -897,8 +890,19 @@ private void DoCloseHelper()
             //Raise Event
             RaiseRunspaceStateEvents();
 
-            // Report telemetry if we have no more open runspaces.
+            if (closeAllOpenRunspaces)
+            {
+                foreach (Runspace runspace in RunspaceList)
+                {
+                    if (runspace.RunspaceStateInfo.State == RunspaceState.Opened)
+                    {
+                        runspace.Dispose();
+                    }
+                }
+            }
 
+            // Report telemetry if we have no more open runspaces.
+            // TODO: Is this still needed?
```

Comment:
This is needed. The old telemetry code is kept as references for the later rewriting work.

Reply:
I'll remove the comment.